sdk/search/azure-search-documents/azure/search/documents/_service/_utils.py

```diff
@@ -151,3 +159,28 @@ def listize_synonyms(synonym_map):
     # type: (dict) -> dict
     synonym_map["synonyms"] = synonym_map["synonyms"].split("\n")
     return synonym_map
+
+def get_access_conditions(model, match_condition=MatchConditions.Unconditionally):
+    # type: (Any, MatchConditions) -> Tuple[Dict[int, Any], AccessCondition]
+    error_map = {
+        401: ClientAuthenticationError,
+        404: ResourceNotFoundError
+    }
+
+    if not hasattr(model, 'e_tag'):
```

Comment:
This does not look correct. If I pass in a model without an etag but specify IfPresent, then this would silently ignore the condition...

Reply:
will still raise it below -

Reply:
This method is returning on line 171 if the attribute is not there. Are you assuming that the caller (directly or indirectly) raises the exception?

Reply:
yes - i can force  a raise even if not so - somethinglike this
```suggestion
    try:
        _unicode_type = unicode
    except NameError:
        _unicode_type = str
    if isinstance(model, _unicode_type):
```

Reply:
also, we have documented that model should be passed to use access conditions - we can perhaps document that passing a string will unconditionally perform the operation and leave this as is

Reply:
Let's explicitly raise an exception if the user passed in an invalid set of arguments. Also, should this be `isinstance(model, six.string_types)`?

Reply:
fiexd

Reply:
explicitly raising if user passes a string and a condition.